features/behat-steps.feature

```diff
@@ -0,0 +1,741 @@
+
+  @require-wp @require-php-server
+  Scenario: Test PHP built-in web server with subdirectory
+    Given an empty directory
+    And an empty wordpress directory
+    And a WP installation in 'wordpress'
+    And a PHP built-in web server to serve 'wordpress'
+    Then the HTTP status code should be 200
+
+  Scenario: Test STDOUT should be empty
+    When I run `echo -n ""`
+    Then STDOUT should be empty
+
+  Scenario: Test running command from subdirectory
+    Given an empty directory
+    And an empty testdir directory
+    When I run `pwd` from 'testdir'
+    Then STDOUT should contain:
+      """
+      testdir
+      """
+
+  Scenario: Test STDOUT strictly contain
+    When I run `echo "exact"`
+    Then STDOUT should strictly contain:
+      """
+      exact
+      """
+
+  Scenario: Test STDERR output
+    When I try `bash -c 'echo "error message" >&2'`
+    Then STDERR should contain:
+      """
+      error message
+      """
+    And STDERR should not be empty
+
+  Scenario: Test file path with RUN_DIR variable
+    Given an empty directory
+    And a {RUN_DIR}/nested.txt file:
+      """
+      content
+      """
+    Then the nested.txt file should exist
+    And the {RUN_DIR}/nested.txt file should contain:
+      """
+      content
+      """
+
+  Scenario: Test multiple files in directory
+    Given an empty directory
+    And a file-a.txt file:
+      """
+      A
+      """
+    And a file-b.txt file:
+      """
+      B
+      """
+    And a file-c.txt file:
+      """
+      C
+      """
+    Then the {RUN_DIR} directory should contain:
+      """
+      file-a.txt
+      """
+    And the {RUN_DIR} directory should contain:
+      """
+      file-b.txt
+      """
+    And the {RUN_DIR} directory should contain:
+      """
+      file-c.txt
+      """
+
+  Scenario: Test special characters in file content
+    Given an empty directory
+    And a special.txt file:
+      """
+      Line with "quotes"
+      Line with 'apostrophes'
+      Line with $variable
+      """
+    Then the special.txt file should contain:
+      """
+      quotes
+      """
+
+  Scenario: Test version comparison operators
+    When I run `echo "5.6.2"`
+    Then STDOUT should be a version string > 5.6.1
+    And STDOUT should be a version string >= 5.6.2
+    And STDOUT should be a version string < 5.6.3
+    And STDOUT should be a version string <= 5.6.2
+    And STDOUT should be a version string == 5.6.2
+    And STDOUT should be a version string != 5.6.3
+
+  @require-wp
+  Scenario: Test JSON array containing
+    Given a WP installation
+    When I run `wp eval 'echo json_encode(["apple", "banana", "cherry"]);'`
+    Then STDOUT should be a JSON array containing:
+      """
+      ["apple", "banana"]
+      """
+
+  @require-wp
+  Scenario: Test email sending detection
+    Given a WP installation
+    When I run `wp user create testuser test@example.com --role=subscriber`
+    Then an email should not be sent
```

Comment:
Since `wp user` is not available, maybe simply require a simple PHP script that calls `wp_mail()` to trigger an email.